xclim/indices/generic.py

```diff
@@ -945,7 +945,7 @@ def get_zones(
         Size of zones
     zone_1 : Quantified | None
         Input value whose zone should be labelled as "zone 1". It should be in the range [`zone_min`, `zone_max` [.
-        By default this is set to `None`, with "zone 1" starting `zone_min`
+        By default this is set to `None`, with "zone 1" aligned with `zone_min`
     bins : Quantified | None
```

Comment:
Isn't the same?

Reply:
It should be the same as a threshold in other indices e.g. "70 degC" and `Quantified` seems to be used in this context

Reply:
Oh geez, maybe. I keep getting confused by that variable type. Hold on.

Reply:
Oh sorry! I'm speaking specifically to `bins`!

I can see that `Quantified` is `xr.DataArray | str | Quantity`, but if we want to provide the option for users setting their own `bins`, this would be provided as `list[str | Quantitiy]`, no?

Reply:
I suggest typing as `xr.DataArray` because it has to be a sequence. `Quantified` is for values that have units and can be of any dimensionality (scalar, vector, matrix, tensor, etc). Usually, when its not a scalar, the dimensions should not have any effect on the computation (like it should not be `time` in a normal indicator). 

But here, accepting `list[Quantified]` kinda means accepting a list of strings with different units. Sticking to an optional DataArray clears the possible confusion.

Also, "Optional[DataArray]" is a xclim-known type, which is good if it ever appears in an indicator's signature.

Reply:
I was the one confused after all!

What I had in mind for bins is really xr.DataArray | list : 
1. xr.DataArray([-60,-55, ... , 70], attrs={"units":"degF"})
2. [-60,-55, ... , 70].  units are the same as `da`

The code could easily accomodate  list[Quantity] e.g. ["5 degC", "10 degC", "10 degC", "20 degC"], but maybe we don't want that as you said. We do accept `zone_min` and `zone_max` as strings with different units though. Should we restrict this?

Reply:
I think we're using `Quantity` and `Quantified` interchangeably here. I think having a list of bins with different units should be acceptable (or, at the very least, supported).

My understanding here is that we can potentially accept the following:
1. `xr.DataArray`
2. `list[str]` -> ["5 degC", "10 degC", "30 degF", "20 K"]
3. `list[pint.Quantity]` -> `[5 * pint.ureg.kelvin, 5 * pint.ureg.celsius]`

I hate the 3rd option, but my understanding is that it is _technically_ supported by `convert_units_to()`. Is this correct?

Reply:
I think keeping `zone_min`, `zone_max` makes sense, invoke them as we do with thresholds, allow different units. 

For bins, xr.DataArray makes more sense. List would be a nice shorthand, but not essential. list[Quantity] seems like a generalization of `zone_min` , `zone_max`, `zone_step` to [`zone_a`, `zone_b`, ..., `zone_z`], but at this point, I'm not sure if it could benefit anybody. I think most people would use: `xr.DataArray([value_a,value_b, ...,], attrs={"units":units})`

Reply:
I agree. I think it's probably enough to support `zone_min` and `zone_max` with different units, but we should ask that `bins` consistently use the same unit. It's very difficult to imagine cases where a user would want something more than that.

We can let them open up an issue if ever that happens, which will likely be "never" haha

Reply:
Ok, let me check what is done elsewhere in the code. I just have the feeling this would be much more flexible than what is otherwise allowed elsewhere, but maybe it's not comparable

Reply:
Ok, implemented `bins : xr.DataArray | list[Quantity] | None`. 

I agree now that a simple list without units was not a good idea. 

As for `xr.DataArray` vs. ` xr.DataArray | list[Quantity]`, I think there is merit to both. We can build indicators with `Optional[xr.DataArray]` and use bins in this way later. Once it's converted, I sort the "list[Quantity]", assuming a user could input a list of bins not necessarily ordered. I can revert changes if we don't agree on this.